src/lsptoolshost/profiling/profiling.ts

```diff
@@ -24,3 +30,271 @@ export function getProfilingEnvVars(outputChannel: LogOutputChannel): NodeJS.Pro
 
     return profilingEnvVars;
 }
+
+export function registerTraceCommand(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    outputChannel: vscode.LogOutputChannel
+): void {
+    context.subscriptions.push(
+        vscode.commands.registerCommand('csharp.recordTrace', async () => {
+            await vscode.window.withProgress(
+                {
+                    location: vscode.ProgressLocation.Notification,
+                    title: 'dotnet-trace',
+                    cancellable: true,
+                },
+                async (progress, token) => {
+                    progress.report({
+                        message: vscode.l10n.t({
+                            message: 'Initializing dotnet-trace...',
+                            comment: 'dotnet-trace is a command name and should not be localized',
+                        }),
+                    });
+                    try {
+                        await executeDotNetTraceCommand(languageServer, progress, outputChannel, token);
+                    } catch (error) {
+                        const errorMessage = error instanceof Error ? error.message : String(error);
+                        showErrorMessageWithOptions(
+                            vscode,
+                            vscode.l10n.t({
+                                message: 'Failed to execute dotnet-trace command: {0}',
+                                args: [errorMessage],
+                                comment: 'dotnet-trace is a command name and should not be localized',
+                            }),
+                            { modal: true }
```

Comment:
Why modal?

Reply:
errors aren't expected - its only if running something failed, so we show a model dialog indicating the command failed.

This is an explicit user requested action, so a much more noticeable modal dialog seems better than an easily ignorable error toast.